### PR TITLE
Revised Lesson204 for ADC Provider

### PR DESCRIPTION
**Pull request by DavidShoe**
We will leave the V1 project up for people to reference but this V2 will replace it in the flow for new people.

### CODE REVIEW

File: Lesson_204V2/StartSolution/MainPage.xaml.cs [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+                CurrentState = newState;
+            }
+        }
+
+        private async Task TextToSpeech(String textToSpeak)
+        {
+            Debug.WriteLine(String.Format("MainPage::TextToSpeech {0}", textToSpeak));
+            // Because we are running somewhere other than the UI thread and we need to talk to a UI element (the media control)
+            // we need to use the dispatcher to move the calls to the right thread.
+            await Dispatcher.RunAsync(
```
Comment (JimGaleForce) [inline Lesson_204V2/StartSolution/MainPage.xaml.cs]: empty - on purpose?

File: Lesson_204V2/FullSolution/Lesson_204V2.csproj [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+      <SubType>Designer</SubType>
+    </Page>
+  </ItemGroup>
+  <ItemGroup>
+    <SDKReference Include="WindowsIoT, Version=10.0.10240.0">
+      <Name>Windows IoT Extensions for the UWP</Name>
+    </SDKReference>
+  </ItemGroup>
+  <ItemGroup>
+    <ProjectReference Include="C:\Users\dashoe\Downloads\BusProviders-develop\BusProviders-develop\ADC\AdcMcp3008\AdcMcp3008.csproj">
```
Comment (ooeygui) [inline Lesson_204V2/FullSolution/Lesson_204V2.csproj]: This has a hard link to your local system

File: Lesson_204V2/StartSolution/Lesson_204V2.csproj [hunk 3]
```diff
@@ -0,0 +1,146 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<Project ToolsVersion="14.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <Import Project="$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\Microsoft.Common.props" Condition="Exists('$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\Microsoft.Common.props')" />
+  <PropertyGroup>
+    <Configuration Condition=" '$(Configuration)' == '' ">Debug</Configuration>
+    <Platform Condition=" '$(Platform)' == '' ">x86</Platform>
+    <ProjectGuid>{C1E13073-C8E5-4C94-A559-15B5C7203BA5}</ProjectGuid>
+    <OutputType>AppContainerExe</OutputType>
```
Comment (ooeygui) [inline Lesson_204V2/StartSolution/Lesson_204V2.csproj]: Make sure the project GUIs is unique (I.e. You didn't clone another project)

**Comment by ooeygui** [comment]
I'm not a fan of versioning this way. It confuses me - what is unique about it? Why did you leave v1?

I'd recommend using a release tag instead of a project fork.

**Comment by ooeygui** [comment]
can't pull this with the local drive reference.
I'd like to not fork it unless seriously compelling reason.

File: Lesson_204V2/FullSolution/.gitignore [hunk 4]
```diff
@@ -0,0 +1,212 @@
+## Ignore Visual Studio temporary files, build results, and
```
Comment (HerrickSpencerMSFT) [inline Lesson_204V2/FullSolution/.gitignore]: Is there a reason to provide a gitignore here? shouldn't the gitignore at the base of repo cover this?
(same with gitattributes above)

File: Lesson_204V2/StartSolution/MainPage.xaml.cs (hunk 1 above)
Comment (anthony-ngu) [inline Lesson_204V2/StartSolution/MainPage.xaml.cs]: yup this is part of the empty solution that people are required to fill in.

File: Lesson_204V2/FullSolution/MainPage.xaml.cs [hunk 5]
```diff
@@ -0,0 +1,196 @@
...
+        {
+            // Checks for state changes and does something when one is detected.
+            if (newState != CurrentState)
+            {
+                String whatToSay;
+
+                switch (newState)
+                {
+                    case eState.JustRight:
+                        {
```
Comment (HerrickSpencerMSFT) [inline Lesson_204V2/FullSolution/MainPage.xaml.cs]: nit: unnecessary brackets in switch cases

File: Lesson_204V2/FullSolution/MainPage.xaml.cs [hunk 6]
```diff
@@ -0,0 +1,196 @@
...
+        const byte CDSADCChannel = 2;
+
+        // Some strings to let us know the current state.
+        const string JustRightLightString = "Ah, just right";
+        const string LowLightString = "I need a light";
+        const string HighLightString = "I need to wear shades";
+
+        // Some internal state information
+        enum eState { unknown, JustRight, TooBright, TooDark};
+        eState CurrentState = eState.unknown;
```
Comment (neilshipp) [inline Lesson_204V2/FullSolution/MainPage.xaml.cs]: Are these public or private?  Below you clearly declare synthesizer as private.  Why not adcController and the AdcChannels?

File: Lesson_204V2/FullSolution/MainPage.xaml.cs (hunk 5 above)
Comment (ooeygui) [inline Lesson_204V2/FullSolution/MainPage.xaml.cs]: Explicit brackets are a recommended coding practice.

**Comment by ooeygui** [comment]
Closing due to update

**Pull request closed without merging**